File: clients/java/Plant.java
import com.sun.jna.Pointer;

///Represents a single Plant on the map.
class Plant extends Mappable
{
  public Plant(Pointer p)
  {
    super(p);
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.plants.length; i++)
    {
      if(BaseAI.plants[i].ID == ID)
      {
        ptr = BaseAI.plants[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands

  ///Command to radiate (heal, attack) another Plant.
  boolean radiate(int x, int y)
  {
    validify();
    return (Client.INSTANCE.plantRadiate(ptr, x, y) == 0) ? false : true;
  }

    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.plantGetId(ptr);
  }
  ///X position of the object
  public int getX()
  {
    validify();
    return Client.INSTANCE.plantGetX(ptr);
  }
  ///Y position of the object
  public int getY()
  {
    validify();
    return Client.INSTANCE.plantGetY(ptr);
  }
  ///The owner of this Plant.
  public int getOwner()
  {
    validify();
    return Client.INSTANCE.plantGetOwner(ptr);
  }
  ///The mutation of this Plant. This mutation refers to list of MutationSpecifications.
  public int getMutation()
  {
    validify();
    return Client.INSTANCE.plantGetMutation(ptr);
  }
  ///The current amount health this Plant has remaining.
  public int getRads()
  {
    validify();
    return Client.INSTANCE.plantGetRads(ptr);
  }
  ///The maximum amount of this health this Plant can have
  public int getMaxRads()
  {
    validify();
    return Client.INSTANCE.plantGetMaxRads(ptr);
  }
  ///The maximum range this plant can radiate
  public int getRange()
  {
    validify();
    return Client.INSTANCE.plantGetRange(ptr);
  }
  ///The distance this plant has left to move
  public int getMovementLeft()
  {
    validify();
    return Client.INSTANCE.plantGetMovementLeft(ptr);
  }
  ///The maximum distance this plant can move each turn
  public int getMaxMovement()
  {
    validify();
    return Client.INSTANCE.plantGetMaxMovement(ptr);
  }
  ///The current power of this plant's radiation
  public int getStrength()
  {
    validify();
    return Client.INSTANCE.plantGetStrength(ptr);
  }
  ///The minimum power of this plant's radiation
  public int getMinStrength()
  {
    validify();
    return Client.INSTANCE.plantGetMinStrength(ptr);
  }
  ///The base power of this plant's radiation
  public int getBaseStrength()
  {
    validify();
    return Client.INSTANCE.plantGetBaseStrength(ptr);
  }
  ///The maximum power of this plant's radiation
  public int getMaxStrength()
  {
    validify();
    return Client.INSTANCE.plantGetMaxStrength(ptr);
  }
  ///The current amount of radiation this Plant can have
  public int getStorage()
  {
    validify();
    return Client.INSTANCE.plantGetStorage(ptr);
  }
  ///The maximum amount of radiation this Plant can have
  public int getMaxStorage()
  {
    validify();
    return Client.INSTANCE.plantGetMaxStorage(ptr);
  }
  ///The number of spores required to spawn this unit
  public int getSpores()
  {
    validify();
    return Client.INSTANCE.plantGetSpores(ptr);
  }

}

File: clients/csharp/Plant.cs
using System;
using System.Runtime.InteropServices;

/// <summary>
/// Represents a single Plant on the map.
/// </summary>
public class Plant: Mappable
{

  public Plant()
  {
  }

  public Plant(IntPtr p)
  {
    ptr = p;
    ID = Client.plantGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public override bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.plants.Length; i++)
    {
      if(BaseAI.plants[i].ID == ID)
      {
        ptr = BaseAI.plants[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

  #region Commands
  /// <summary>
  /// Allows a plant to display messages on the screen
  /// </summary>
  public bool talk(string message)
  {
    validify();
    return (Client.plantTalk(ptr, message) == 0) ? false : true;
  }
  /// <summary>
  /// Command to radiate (heal, attack) another Plant.
  /// </summary>
  public bool radiate(int x, int y)
  {
    validify();
    return (Client.plantRadiate(ptr, x, y) == 0) ? false : true;
  }
  /// <summary>
  /// Command to uproot a plant and move to a new location.
  /// </summary>
  public bool uproot(int x, int y)
  {
    validify();
    return (Client.plantUproot(ptr, x, y) == 0) ? false : true;
  }
  #endregion

  #region Getters
  /// <summary>
  /// Unique Identifier
  /// </summary>
  public new int Id
  {
    get
    {
      validify();
      int value = Client.plantGetId(ptr);
      return value;
    }
  }

  /// <summary>
  /// X position of the object
  /// </summary>
  public new int X
  {
    get
    {
      validify();
      int value = Client.plantGetX(ptr);
      return value;
    }
  }

  /// <summary>
  /// Y position of the object
  /// </summary>
  public new int Y
  {
    get
    {
      validify();
      int value = Client.plantGetY(ptr);
      return value;
    }
  }

  /// <summary>
  /// The owner of this Plant.
  /// </summary>
  public int Owner
  {
    get
    {
      validify();
      int value = Client.plantGetOwner(ptr);
      return value;
    }
  }

  /// <summary>
  /// The mutation of this Plant. This mutation refers to list of MutationSpecifications.
  /// </summary>
  public int Mutation
  {
    get
    {
      validify();
      int value = Client.plantGetMutation(ptr);
      return value;
    }
  }

  /// <summary>
  /// The current amount of radiation this Plant has.
  /// </summary>
  public int Rads
  {
    get
    {
      validify();
      int value = Client.plantGetRads(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum amount of radiation this Plant can have before dying.
  /// </summary>
  public int MaxRads
  {
    get
    {
      validify();
      int value = Client.plantGetMaxRads(ptr);
      return value;
    }
  }

  /// <summary>
  /// The remaining number of times the plant can radiate.
  /// </summary>
  public int RadiatesLeft
  {
    get
    {
      validify();
      int value = Client.plantGetRadiatesLeft(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times the plant can radiate.
  /// </summary>
  public int MaxRadiates
  {
    get
    {
      validify();
      int value = Client.plantGetMaxRadiates(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum range this plant can radiate.
  /// </summary>
  public int Range
  {
    get
    {
      validify();
      int value = Client.plantGetRange(ptr);
      return value;
    }
  }

  /// <summary>
  /// The remaining number of times this plant can be uprooted.
  /// </summary>
  public int UprootsLeft
  {
    get
    {
      validify();
      int value = Client.plantGetUprootsLeft(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times this plant can be uprooted.
  /// </summary>
  public int MaxUproots
  {
    get
    {
      validify();
      int value = Client.plantGetMaxUproots(ptr);
      return value;
    }
  }

  /// <summary>
  /// The current power of this plant.
  /// </summary>
  public int Strength
  {
    get
    {
      validify();
      int value = Client.plantGetStrength(ptr);
      return value;
    }
  }

  /// <summary>
  /// The minimum power of this plant.
  /// </summary>
  public int MinStrength
  {
    get
    {
      validify();
      int value = Client.plantGetMinStrength(ptr);
      return value;
    }
  }

  /// <summary>
  /// The base power of this plant.
  /// </summary>
  public int BaseStrength
  {
    get
    {
      validify();
      int value = Client.plantGetBaseStrength(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum power of this plant.
  /// </summary>
  public int MaxStrength
  {
    get
    {
      validify();
      int value = Client.plantGetMaxStrength(ptr);
      return value;
    }
  }

  #endregion

  #region Properties
  #endregion
}
